spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-governance-routing/src/main/java/com/alibaba/cloud/routing/cache/GlobalInstanceStatusListCache.java

```diff
@@ -0,0 +1,283 @@
+			instanceInfoMap.put(instance.getIp() + ":" + instance.getPort(), sif);
+			serviceInstanceList.add(instanceInfoMap);
+			Map<String, List<Map<String, ServiceInstanceInfo>>> serviceMap = new ConcurrentHashMap<>();
+			serviceMap.put(targetName, serviceInstanceList);
+			globalServiceList.add(serviceMap);
+		}
+		else {
+			for (Map<String, List<Map<String, ServiceInstanceInfo>>> serviceMap : globalServiceList) {
+				for (String key : serviceMap.keySet()) {
+					if (Objects.equals(key, targetName)) {
+
+						List<Map<String, ServiceInstanceInfo>> serviceInstanceList = serviceMap
+								.get(targetName);
+						Map<String, ServiceInstanceInfo> infoHashMap = new HashMap<>();
+						infoHashMap.put(instance.getIp() + ":" + instance.getPort(), sif);
+						serviceInstanceList.add(infoHashMap);
+
+						serviceMap.replace(targetName, serviceInstanceList);
+					}
+					else {
+						List<Map<String, ServiceInstanceInfo>> serviceInstanceList = new ArrayList<>();
+						Map<String, ServiceInstanceInfo> instanceInfoMap = new HashMap<>();
+						instanceInfoMap.put(instance.getIp() + ":" + instance.getPort(),
+								sif);
+						serviceInstanceList.add(instanceInfoMap);
+						Map<String, List<Map<String, ServiceInstanceInfo>>> serviceMapTmp = new HashMap<>();
+						serviceMapTmp.put(targetName, serviceInstanceList);
+						globalServiceList.add(serviceMapTmp);
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Check instance in cache.
+	 * @param target target.
+	 * @param instance instance object.
+	 * @return boolean
+	 */
+	public static boolean checkContainersInstance(String target, Instance instance) {
+		AtomicBoolean res = new AtomicBoolean(false);
+
+		globalServiceList.forEach(val -> val.forEach((var9, var10) -> {
+			if (val.containsKey(target)) {
+				var10.forEach(var1 -> {
+					if (var1.containsKey(instance.getIp() + ":" + instance.getPort())) {
+
+						res.set(true);
+					}
+				});
+			}
+		}));
+
+		return res.get();
+	}
+
+	/**
+	 * Return GlobalCache Object.
+	 * @return return global instance cache
+	 */
+	public static List<Map<String, List<Map<String, ServiceInstanceInfo>>>> getAll() {
+
+		return globalServiceList;
+	}
+
+	/**
+	 * Get service instance list by service name.
+	 * @param targetServiceName service name.
+	 * @return instance list.
+	 */
+	public static List<Map<String, ServiceInstanceInfo>> getInstanceByServiceName(
+			String targetServiceName) {
+
+		for (Map<String, List<Map<String, ServiceInstanceInfo>>> map : globalServiceList) {
+			if (map.containsKey(targetServiceName)) {
+				return map.get(targetServiceName);
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Get instance info by global cache.
+	 * @param instanceName instance name
+	 * @return sif object
+	 */
+	public static ServiceInstanceInfo getInstanceByInstanceName(String instanceName) {
+
+		for (Map<String, List<Map<String, ServiceInstanceInfo>>> map : globalServiceList) {
+			for (String serviceName : map.keySet()) {
```

Comment:
Use lambda expression, it's more clear. Such as `instanceInfoMap.forEach()` and check your all codes.

Reply:
done!